backend/api/professor/grading_materials.py

```
@@ -0,0 +1,333 @@
+            )
+        
+        # Validate file size (max 10MB)
+        file_size = 0
+        content = await file.read()
+        file_size = len(content)
+        
+        if file_size > 10 * 1024 * 1024:  # 10MB
+            raise HTTPException(
+                status_code=400,
+                detail=f"File {file.filename} too large. Maximum size is 10MB"
+            )
```

Ruff (0.14.1)

Abstract `raise` to an inner function

(TRY301)